reconcile/openshift_base.py

```diff
@@ -556,18 +570,441 @@ def check_unused_resource_types(ri):
+            "cluster": cluster,
+            "namespace": namespace,
+            "kind": resource_type,
+            "name": name,
+            "privileged": options.privileged,
+        }
+        actions.append(action)
+        apply_action(
+            oc_map=oc_map,
+            ri=ri,
+            cluster=cluster,
+            namespace=namespace,
+            resource_type=resource_type,
+            resource=desired,
+            options=options,
+        )
+
+    return actions
+
+
+def handle_modified_resources(
+    oc_map: OC_Map,
+    ri: ResourceInventory,
+    modified_resources: Mapping[Any, Any],
+    cluster: str,
+    namespace: str,
+    resource_type: str,
+    data: Mapping[Any, Any],
+    options: ApplyOptions,
+) -> list[dict[str, Any]]:
+    actions: list[dict[str, Any]] = []
+
+    for name, dp in modified_resources.items():
+        b_apply = True
+        current: OR = dp.current
+        desired: OR = dp.desired
+
+        if not options.dry_run and options.no_dry_run_skip_compare:
+            msg = (
+                f"[{cluster}/{namespace}] skipping compare of resource"
+                f"'{resource_type}/{name}'"
+            )
+            logging.debug(msg)
+        else:
+            if (
+                not options.take_over
+                and options.caller
+                and current.caller != options.caller
+                and options.all_callers
+                and current.caller in options.all_callers
+            ):
+                ri.register_error()
+                logging.error(
+                    f"[{cluster}/{namespace}] resource '{resource_type}/{name}' present and managed by another caller: {current.caller}"
+                )
+                b_apply = False
+            else:
+                logging.debug("CURRENT: " + OR.serialize(OR.canonicalize(current.body)))
+
+        if b_apply:
+            options.privileged = data["use_admin_token"].get(name, False)
+            action = {
+                "action": ACTION_APPLIED,
+                "cluster": cluster,
+                "namespace": namespace,
+                "kind": resource_type,
+                "name": name,
+                "privileged": options.privileged,
+            }
+            actions.append(action)
+            apply_action(
+                oc_map=oc_map,
+                ri=ri,
+                cluster=cluster,
+                namespace=namespace,
+                resource_type=resource_type,
+                resource=desired,
+                options=options,
+            )
+    return actions
+
+
+def handle_deleted_resources(
+    oc_map: OC_Map,
+    ri: ResourceInventory,
+    deleted_resources: Mapping[Any, Any],
+    cluster: str,
+    namespace: str,
+    resource_type: str,
+    data: Mapping[Any, Any],
+    options: ApplyOptions,
+) -> list[dict[str, Any]]:
+    actions: list[dict[str, Any]] = []
+
+    for name, current in deleted_resources.items():
+        b_delete = True
```

Comment:
similar here, can extract a method for `should_delete`

Reply:
done.